### DISCUSSION TITLE
Question about error code

### DISCUSSION THREAD

**Discussion by gittwe** [Sleepw4lker/TameMyCerts#15] [Q&A] [answered]
Hi,

When enrolling for a certificate I always get the following error message from TameMyCerts:
`Unable to parse the given certificate request. Request type was 1024.`

I have increased the log level of the CA but still get no more information. Is there a list of error codes I could lookup what this message means?

The csr contain s a rsa 2048 bit key.

Thanks,
Thomas

**Comment by Sleepw4lker** [comment]
Hello, and thanks for reaching out. A request type of 1024 (0x400) indicates that this should be a CMC encoded request.

Can you provide more details on how the request was created and submitted?

Furthermore, can you inspect the request (from "Failed Requests" in die ADCS MMC, export it and run certutil -dump against the file) to ensure it contains a PKCS#10 certificate request?

Cheers

**Comment by gittwe** [comment]
Hi,

Thank for getting back on this so fast.

Indeed the request is not CMC encoded, it is a P10 request sent to the CA using certreq.exe. It is handled by the CA without any issue using the Default policy module.

Does TameMyCerts supports P10 encoded csr?

**Comment by gittwe** [comment]
The enrollment goes over the CES service:
certreq.exe  -f -kerberos -config <CES Enrollment service URL> -submit -attrib CertificateTemplate:TestTemplate test01.csr test01.cer

**Comment by gittwe** [comment]
Here is the output of certutil -dump:

PKCS10 Certificate Request:
Version: 1
Subject:
    E=test@example.com
    CN=test01.example.com
  Name Hash(sha1): cd2ec3eb6877b4832365a467a74df1d2e243bfdd
  Name Hash(md5): bd3dcf25afefa6384d8199309c08befc

Public Key Algorithm:
    Algorithm ObjectId: 1.2.840.113549.1.1.1 RSA (RSA_SIGN)
    Algorithm Parameters:
    05 00
Public Key Length: 2048 bits
Public Key: UnusedBits = 0
    0000  30 82 01 0a 02 82 01 01  00 ca 33 a2 98 d7 86 77
    0010  53 07 ef e3 df 4d d5 49  80 84 d9 fc 0b 58 2b 34
    0020  b3 bf 46 e0 fd 5f 3c 15  1b 57 9d 18 3c ee 37 a3
    0030  78 b1 5d 88 62 61 ae f0  ee 26 09 77 c0 08 d6 e2
    0040  4b c8 4b 8c 15 1e 0c f2  be f8 4b ff 2f 76 04 a7
    0050  ab fe 76 7d ab 00 49 eb  5d a9 c5 fb a8 22 8e 2a
    0060  54 cf 29 06 22 fa 71 2f  fa 34 b3 dd 30 f5 e9 ed
    0070  b3 45 1e e7 50 e4 03 4c  26 4c 22 44 a4 ec 4b 60
    0080  1b 43 0e 4c bd b4 1a b6  3a 61 10 21 a6 6b 9f f8
    0090  8d 93 27 a5 59 e3 7a b2  d1 4e f1 9d 74 9c ba 59
    00a0  d5 82 b3 1e 25 03 ee bc  e4 ec ce fa f1 8c 5e 77
    00b0  f3 0e 26 27 58 b9 a7 5d  03 00 b6 fd 1c cc 5d 99
    00c0  b8 26 00 47 0c b8 07 99  ff f3 ec 36 2a 68 07 32
    00d0  e7 ab f3 a9 cd 20 47 d2  33 ae 7c 56 cc 6c f7 35
    00e0  19 49 03 59 80 3f 1c 55  5e 68 4a 75 42 91 b6 02
    00f0  a9 fa c7 11 4c f2 2a 80  81 07 4f bc 9d 26 64 fb
    0100  9d 52 fe d3 6e 69 d1 c3  b3 02 03 01 00 01
Request Attributes: 0
  0 attributes:
Signature Algorithm:
    Algorithm ObjectId: 1.2.840.113549.1.1.11 sha256RSA
    Algorithm Parameters:
    05 00
Signature: UnusedBits=0
    0000  5d dd 1b aa 50 71 14 c0  21 af 00 e5 14 d5 96 fd
    0010  16 c6 58 50 4c b4 e1 66  e0 61 ca c1 d2 f2 8c 6d
    0020  23 16 14 3e cf 1a ab 17  2d 9a 5c e5 c9 51 6e 0c
    0030  ad 05 51 9d c7 1b ca cf  08 64 7f bd 6b e3 6b a6
    0040  55 0d 1a 5c 4e 53 fd 63  2d 2e 59 7c 2c 1a c0 50
    0050  b0 53 45 d3 7a 99 c6 6f  da 8b af f8 ac a5 2b 87
    0060  07 56 b6 45 ea df 3f cd  49 22 e9 97 ef 07 e7 8d
    0070  33 74 2c b3 b5 54 2a 68  b6 d2 5d fe f1 3e 42 da
    0080  a1 53 c2 4a 83 8d 2e b4  95 91 b3 de 7b d1 32 8b
    0090  05 a8 ae c9 3c dc d1 1b  4b b3 8c 2b 37 4a d8 a0
    00a0  cd 9b 00 fd 74 0c 8e 7d  c8 29 fa 35 6d 61 7f 3f
    00b0  63 85 b3 0a b6 e4 9f 99  65 8d 30 9d 70 78 36 79
    00c0  8f be 3c 58 5b cf 24 68  79 4a 59 a2 0a a3 0d 7c
    00d0  ec 0b c7 db 6c bf 03 ac  ef db 27 7e ed 05 c6 ea
    00e0  10 cc 8e 27 e4 de 77 d1  86 d3 81 d5 85 3f 64 1b
    00f0  0e da b1 1b 13 85 6f d4  a6 71 dd 6f b0 16 42 b7
Signature matches Public Key
Key Id Hash(rfc-sha1): 4df20b27ff787902d18a9643878654a58e3a6115
Key Id Hash(sha1): 368b167671dd7ed75f5f1dee5003972eb00125e5
Key Id Hash(bcrypt-sha1): 772f70e974b2086bbbf5c1423e12c6c62b062b15
Key Id Hash(bcrypt-sha256): c257be9c3c62e6372fad0d409450785d8ae9e40a40706460e33a0f18d4731189
CertUtil: -dump command completed successfully.

**Comment by Sleepw4lker** [comment]
Hello @gittwe . 

How was this request created? Any chance you can share the originating command, if any? (so that I can reproduce it)
Does it work when directly submitting to the CA via RPC/DCOM?
You also may reach out to me directly if you'd like to set up a remote session with me so that we can have a look at the system together.

Cheers

**Comment by Sleepw4lker** [comment, marked as answer]
Hello @gittwe . Thanks for reaching out to me. Indeed the request ist of type CMC, TameMyCerts (or the underlying library) is not able to properly parse it. I will fix this in the upcoming version by avoiding to parse requests at all. This is a relic from previous versions that did not pull this information directly from the CA process. Thanks for your help to make TameMyCerts even better.
